tracee-ebpf/tracee/net_events.go

```diff
@@ -30,83 +38,40 @@ func (t *Tracee) processNetEvents() {
 
 			evtMeta, dataBuff := parseEventMetaData(in)
 
-			// timeStamp is nanoseconds since system boot time
 			timeStampObj := time.Unix(0, int64(evtMeta.TimeStamp+t.bootTime))
 
+			if t.config.Output.RelativeTime {
+				// To get the current ("wall") time, we add the boot time into it.
+				evtMeta.TimeStamp -= t.startTime
+			} else {
+				// timeStamp is nanoseconds since system boot time
+				// To get the monotonic time since tracee was started, we have to subtract the start time from the timestamp.
+				evtMeta.TimeStamp += t.bootTime
+			}
+
 			if evtMeta.NetEventId == NetPacket {
-				var pktLen uint32
-				err := binary.Read(dataBuff, binary.LittleEndian, &pktLen)
-				if err != nil {
-					t.handleError(err)
-					continue
-				}
-				var ifindex uint32
-				err = binary.Read(dataBuff, binary.LittleEndian, &ifindex)
-				if err != nil {
-					t.handleError(err)
-					continue
-				}
-				idx, ok := t.ngIfacesIndex[int(ifindex)]
-				if !ok {
-					t.handleError(err)
-					continue
+				captureData, err := parseCaptureData(dataBuff)
+				if err == nil && len(t.config.Capture.NetIfaces) > 0 {
+					if err := t.writePacket(captureData, time.Unix(int64(evtMeta.TimeStamp), 0), bytes.NewBuffer(dataBuff)); err != nil {
```

Comment:
Ok
Firstly, I changed it to `payloadBytes`.
Secondly, the reason I return a byte array is because now we aren't need the `bytes.NewBuffer()` anymore (besides in the writePacket and in the debug events which in the next PR will be use the byte array instead the `newBuffer`).
BTW another reason to use a byte array instead of the `newBuffer `is because if in the packet parsing we now using the the `binary.littleendian.uint` function which use a byte array and not a `newBuffer`